### DISCUSSION TITLE
ROCm 5.7.1 Release

### DISCUSSION THREAD

**Discussion by saadrahim** [ROCm/ROCm#2559] [Announcements]
ROCm 5.7.1 is point release with the following changes:

### rocBLAS
A new functionality rocblas-gemm-tune and an environment variable ROCBLAS_TENSILE_GEMM_OVERRIDE_PATH are added to rocBLAS in the ROCm 5.7.1 release.

*rocblas-gemm-tune* is used to find the best-performing GEMM kernel for each GEMM problem set. It has a command line interface, which mimics the --yaml input used by rocblas-bench. To generate the expected --yaml input, profile logging can be used, by setting the environment variable ROCBLAS_LAYER4.

For more information on rocBLAS logging, see [Logging in rocBLAS](https://rocm.docs.amd.com/projects/rocBLAS/en/docs-5.7.1/API_Reference_Guide.html#logging-in-rocblas).

An example input file: Expected output (note selected GEMM idx may differ): Where the far right values (solution_index) are the indices of the best-performing kernels for those GEMMs in the rocBLAS kernel library. These indices can be directly used in future GEMM calls. See rocBLAS/samples/example_user_driven_tuning.cpp for sample code of directly using kernels via their indices.

If the output is stored in a file, the results can be used to override default kernel selection with the kernels found, by setting the environment variable ROCBLAS_TENSILE_GEMM_OVERRIDE_PATH, where points to the stored file.

For more details, refer to the [rocBLAS Programmer's Guide.](https://rocm.docs.amd.com/projects/rocBLAS/en/latest/Programmers_Guide.html#rocblas-gemm-tune)

### HIP 5.7.1 (for ROCm 5.7.1)

ROCm 5.7.1 is a point release with several bug fixes in the HIP runtime.

#### Fixed 
- The *hipPointerGetAttributes* API returns the correct HIP memory type as *hipMemoryTypeManaged* for managed memory.

### hipSOLVER 1.8.2

hipSOLVER 1.8.2 for ROCm 5.7.1

#### Fixed

- Fixed conflicts between the hipsolver-dev and -asan packages by excluding
  hipsolver_module.f90 from the latter

<hr /><em>This discussion was created from the release <a href='https://github.com/RadeonOpenCompute/ROCm/releases/tag/rocm-5.7.1'>ROCm 5.7.1 Release</a>.</em>

**Comment by kyflores** [comment]
This is great news, and thanks for the update. Can you clarify if this release adds official support for the 7900XTX and W7900? [This AMD community blog post](https://community.amd.com/t5/ai/amd-extends-support-for-pytorch-machine-learning-development-on/ba-p/637756) community implies that they're now supported, at least in PyTorch, and [Phoronix ran an article about it](https://www.phoronix.com/news/RX-7900-XTX-ROCm-PyTorch), but I don't see either of these devices mentioned in the [support table](https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html).

**Reply by johnnynunez** [reply]
> This is great news, and thanks for the update. Can you clarify if this release adds official support for the 7900XTX and W7900? [This AMD community blog post](https://community.amd.com/t5/ai/amd-extends-support-for-pytorch-machine-learning-development-on/ba-p/637756) community implies that they're now supported, at least in PyTorch, and [Phoronix ran an article about it](https://www.phoronix.com/news/RX-7900-XTX-ROCm-PyTorch), but I don't see either of these devices mentioned in the [support table](https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html).

See documentation. Now, there is radeon section installation

**Comment by husmen** [comment]
> > This is great news, and thanks for the update. Can you clarify if this release adds official support for the 7900XTX and W7900? [This AMD community blog post](https://community.amd.com/t5/ai/amd-extends-support-for-pytorch-machine-learning-development-on/ba-p/637756) community implies that they're now supported, at least in PyTorch, and [Phoronix ran an article about it](https://www.phoronix.com/news/RX-7900-XTX-ROCm-PyTorch), but I don't see either of these devices mentioned in the [support table](https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html).
> 
> See documentation. Now, there is radeon section installation

Let me rephrase @kyflores's question:
- The [documentation](https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html#linux-supported-gpus) only lists support for "AMD Radeon VII" on Linux.
- This [AMD blog post](https://community.amd.com/t5/ai/amd-extends-support-for-pytorch-machine-learning-development-on/ba-p/637756) claims support for "AMD Radeon RX 7900 XTX" on Linux.

So, is the blog post mistaken? Or is the documentation incomplete?

**Reply by saadrahim** [reply]
I updated the documentation support tables. https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html

**Comment by xiaobo1025** [comment]
Hello, my friend, can you help me with how to install ROCm? 
Thank you very much!

**Reply by johnnynunez** [reply]
```bash
sudo apt-get update && apt-get install -y git vim curl build-essential python3-pip python3-venv libpython3-dev libpng-dev libjpeg-dev
sudo apt-get update && sudo apt-get install -y \
  build-essential \
  clang \
  clang-format \
  clang-tidy \
  cmake
curl -O https://repo.radeon.com/amdgpu-install/6.0/ubuntu/jammy/amdgpu-install_6.0.60000-1_all.deb
dpkg -i amdgpu-install_6.0.60000-1_all.deb

DEBIAN_FRONTEND=noninteractive amdgpu-install --usecase=graphics,rocm -y
sudo usermod -a -G render,video $LOGNAME
```

**Reply by xiaobo1025** [reply]
@johnnynunez Hello,my friend ,I want to ask that could I deploy ROCm on my windows 10 system?I see that the official website of ROCm provides a guide on deploying ROCm SDK on windows 10 systems. Can I deploy ROCm and merge ROCm on my windows 10 systems? 
I really look forward to your reply, my dear friend.

**Comment by xuhuisheng** [comment]
@saadrahim 
Thank you for releasing ROCm-5.7.1. ROCm said rocm-5.7.x is the last version for supporting MI50, so I am really appreciate for make rocm-5.7.x branch more stable.

So is there a ROCm-5.7.3 version release? Thank you.

**Comment by briansp2020** [comment]
https://repo.radeon.com/amdgpu/5.7.3/
https://repo.radeon.com/rocm/apt/5.7.3/
